enterprise/server/backends/pebble_cache/pebble_cache.go

```diff
@@ -1538,6 +1567,17 @@ func (p *PebbleCache) findMissing(ctx context.Context, db pebble.IPebbleDB, r *r
 			return err
 		}
 	}
+
+	// If this is a GCS object, ensure the custom time is relatively recent
+	// so that we avoid saying something exists when it's been deleted by
+	// a GCS lifecycle rule.
+	if gcsMetadata := md.GetStorageMetadata().GetGcsMetadata(); gcsMetadata != nil {
+		customTime := time.UnixMicro(gcsMetadata.GetLastCustomTimeUsec())
+		if p.clock.Since(customTime) > time.Duration(p.gcsTTLDays)*(time.Hour*24) {
```

Comment:
so if you increase the bucket TTL, do you have to do anything to the values already written to the bucket? the concern would be that a value could look like it's not going to expire when it actually will.

seems like not (i.e., everything's fine) just trying to game out weird edge cases.